client.go

```diff
@@ -304,9 +316,60 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return offset, err
 }
 
+func (client *client) Coordinator(consumerGroup string) (*Broker, error) {
+	client.lock.RLock()
+	coordinator := client.coordinators[consumerGroup]
+	client.lock.RUnlock()
+
+	if coordinator == nil {
+		var err error
+		err = client.RefreshCoordinator(consumerGroup)
+		if err != nil {
+			return nil, err
+		}
+
+		client.lock.RLock()
+		coordinator = client.coordinators[consumerGroup]
+		client.lock.RUnlock()
+	}
+
+	_ = coordinator.Open(client.conf)
+	return coordinator, nil
+}
+
+func (client *client) RefreshCoordinator(consumerGroup string) error {
+	coordinator, err := client.getCoordinator(consumerGroup, client.conf.Metadata.Retry.Max)
+	if err != nil {
+		return err
+	}
+
+	client.lock.Lock()
+	client.coordinators[consumerGroup] = client.registerBroker(coordinator)
```

Comment:
Tangent, but related: I'm an idiot. `ConsumerMetadataResponse` should just have a `*Broker` as a field, it lines up exactly with `MetadataResponse` and the broker already has a `decode` method which does the right thing given an id/host/port tuple. I've added it to the wiki.

Back on topic: I think this change would sit better for me if `getCoordinator` was called `getConsumerMetadata` and just returned the response object on success (letting `RefreshMetadata` construct the broker object).